### ISSUE TITLE
Largest Contentful Paint vs Font preloading

### ISSUE THREAD

**Issue by xoxys** [GoogleChrome/lighthouse#11460]
I'm a bit confused about the recommendation `uses-rel-preload` and the resulting `LCP` score. Without font preloading the `LCP` score looks ok and results in a 100 % rating in the performance category:

![grafik](https://user-images.githubusercontent.com/3391958/93643439-50c12280-fa00-11ea-9980-3583a81a7a99.png)

After adding the suggested preloading lines for my fonts the `LCP` increased a lot and the performance rating goes down to 80 %:

![grafik](https://user-images.githubusercontent.com/3391958/93643534-7bab7680-fa00-11ea-9a2c-0742b06ee6e0.png)

The preloading lines I use looks like this:

```
<link rel="preload" as="font" href="/fonts/Metropolis.woff2" type="font/woff2" crossorigin="anonymous">
<link rel="preload" as="font" href="/fonts/LiberationSans.woff2" type="font/woff2" crossorigin="anonymous">
<link rel="preload" as="font" href="/fonts/LiberationSans-Bold.woff2" type="font/woff2" crossorigin="anonymous">
<link rel="preload" as="font" href="/fonts/LiberationSans-BoldItalic.woff2" type="font/woff2" crossorigin="anonymous">
<link rel="preload" as="font" href="/fonts/LiberationSans-Italic.woff2" type="font/woff2" crossorigin="anonymous">
<link rel="preload" as="font" href="/fonts/LiberationMono.woff2" type="font/woff2" crossorigin="anonymous">
<link rel="preload" as="font" href="/fonts/DroidSans.woff2" type="font/woff2" crossorigin="anonymous">
```
What am I doing wrong? Any way to get preloading and the `LCP` together? Thanks a lot for your help.

**Comment by patrickhulce** [comment]
Thanks for filing @xoxys! Is this page deployed on the internet anywhere for us to take a look? I have a hunch about what's going on, but we need a page to be able to look into it :)

**Comment by xoxys** [comment]
Not right  now :) would you need it with the preloads enabled?

**Comment by patrickhulce** [comment]
> would you need it with the preloads enabled?

Ideally we could have both versions available at the same time to directly compare, but the one with the preloads enabled is the more important of the two to hunt down the (likely) bug.

**Comment by xoxys** [comment]
Ok, here you go:
https://preload-on.geekdocs.de/
https://preload-off.geekdocs.de/

Let me know when you need something else

**Comment by patrickhulce** [comment]
Thanks very much for those @xoxys! I'm having trouble reproducing the low performance on the preload one though :/

![image](https://user-images.githubusercontent.com/2301202/93650582-7e907200-f9d4-11ea-9cc8-233438bcf1d3.png)

Any extra settings, environment information, etc that might help?

**Comment by xoxys** [comment]
Ok now it is getting more complicated :) Ok, so see this behavior while testing on localhost and I'm using the following config:

```
---
ci:
  collect:
    numberOfRuns: 3
    staticDistDir: exampleSite/public
    url:
      - http://localhost/
      - http://localhost/404.html
      - http://localhost/posts/
    settings:
      chromeFlags: "--no-sandbox"
  assert:
    preset: "lighthouse:no-pwa"
    assertions:
      uses-long-cache-ttl: off
      tap-targets: warn
      unsized-images: warn
      categories:performance:
        - error
        - minScore: 0.95
      categories:accessibility:
        - error
        - minScore: 1
      categories:seo:
        - error
        - minScore: 0.95
  upload:
    target: filesystem
    outputDir: lhci_reports
```

If you want to test it locally as well you could download the whole source code (pre-build and working) from the S3 backend which is serving the demo site. https://sp.rknet.org/minio/geekdocs/ (there should be a download button behind the preload-on and preload-off folder). I could also provide a tarball here if that works better for you :)

**Comment by xoxys** [comment]
Oh maybe I should have mentioned that I use `lhci`. Now I am not sure if I opened the issue in the right repo... But I thought this was more of a core problem.

**Comment by patrickhulce** [comment]
You're all good you actually filed in LHCI @xoxys but I moved the issue to the core repo because any issue computing the performance metrics won't be LHCI specific :)

I can repro locally now thanks for the download link 👍 

This is a bug in simulated throttling. Basically the preloaded fonts now load before LCP and Lighthouse includes them in our pessimistic estimate of LCP. It only happens on localhost because they load *so* fast that Chrome doesn't even have a chance to get a paint out. When the page is hosted somewhere remotely we see the expected behavior. 

As a temporary workaround you can use
```yaml
ci:
  collect:
    settings:
      throttlingMethod: devtools
```

but either way it's not a real problem with the performance of the page.

To the Lighthouse team, I think it's time to finally include more than just trace/devtools information in our lantern graph creation. These fonts are all display `swap` and we shouldn't be including them in the LCP calculation.

**Comment by xoxys** [comment]
Wow, many thanks for your help! To be honest the workaround does not really work for me. With `devtools` enabled tthe results changed to

**Comment by xoxys** [comment]
Is there any update so far?

**Comment by patrickhulce** [comment]
No, changes to lantern like this requires are large investments and are only shipped with major versions. There is no immediate plan to tackle this.

**Comment by paulirish** [comment]
We've since "disabled" the preload audit for other, but related reasons. I'll leave this issue open to track patrick's suggestion:

> To the Lighthouse team, I think it's time to finally include more than just trace/devtools information in our lantern graph creation. These fonts are all `font-display: swap` and we shouldn't be including them in the LCP calculation.

**Comment by paulirish** [comment]
We believe this may only affect localhost.. generally a swap font will have deferred network requests.. but in localhost.. all requests are so fast it happens prior to LCP.
Assuming that is the case, this is lower priority. Localhost will always have very optimistic and unrealistic perf. It's something we can try to account for, but we're limited..

**Comment by siakaramalegos** [comment]
FYI I think I'm having this issue on my website in production, not in dev. Right now, I've commented out my preloads. Site is https://sia.codes/
